src/crypto/crypto_mbedtls_alt.c

```diff
@@ -2261,6 +2263,7 @@ int crypto_ec_point_invert(struct crypto_ec *e, struct crypto_ec_point *p)
                0 :
                -1;
 }
+#endif /* CRYPTO_MBEDTLS_CRYPTO_EC */
```

Comment:
previous code from line 1966 #ifdef CRYPTO_MBEDTLS_CRYPTO_EC to line 2945 is all included by this macro.
So why do we split them here? As CRYPTO_MBEDTLS_CRYPTO_EC_DPP is actually inside CRYPTO_MBEDTLS_CRYPTO_EC.
If you want to put some apis out of macro, just move them out.
Adding macros everywhere will make code more complex.

Reply:
Earlier `CRYPTO_MBEDTLS_CRYPTO_EC` was not under any check. Now it is under a set of checks. Earlier, it was as if the functions were all getting in and that particular `#ifdef CRYPTO_MBEDTLS_CRYPTO_EC` for had no role to play. Those APIs were always included with the #define statement for `CRYPTO_MBEDTLS_CRYPTO_EC`.

Now since, `CRYPTO_MBEDTLS_CRYPTO_EC` is under check, I placed them out. Since in this specific case, DPP is also part of that, I can remove the split here i suppose

Reply:
I have removed the intermediate checks. pushing in changes. Kindly review